### ISSUE TITLE
Question about loss=nan.

### ISSUE THREAD

**Issue by yangdaowu** [ws-choi/Conditioned-Source-Separation-LaSAFT#12]
Hello

I want to know when the run reaches 115 epochs with loss=nan, I checked the checkpoints and the last saved ckpt is at the 79th epoch

Using the example you gave python main.py --problem_name conditioned_separation --mode train --run_id lasaft_net --musdb_root etc/musdb18_dev_wav --gpus 1 --precision 16 --batch_size 6 --num_workers 0 --pin_memory True --save_top_k 3 --save_weights_only True --patience 10 --lr 0.001 --model CUNET_TFC_GPoCM_LaSAFT

I await feedback, thank you very much:
![image](https://user-images.githubusercontent.com/64669901/111891336-34dd1e80-8a2d-11eb-80d7-e47fce73cab9.png)

**Comment by ws-choi** [comment]
Hi @yangdaowu,
Was the dataset you used Musdb sample dataset?
Did you download it from [this guide](https://github.com/ws-choi/Conditioned-Source-Separation-LaSAFT/tree/main/lasaft/data)?

**Comment by yangdaowu** [comment]
Yes, I am using the dataset MUSDB18

**Comment by ws-choi** [comment]
I mean, was it the sample dataset of MUSDB18 or the full dataset of it?

**Comment by yangdaowu** [comment]
Is its complete data set
![image](https://user-images.githubusercontent.com/64669901/111896398-4d146400-8a54-11eb-9d9d-f868bdf23bf2.png)

**Comment by yangdaowu** [comment]
This happens when I keep dragging the command line
![image](https://user-images.githubusercontent.com/64669901/111896817-18ee7280-8a57-11eb-8d14-afca54f6af1d.png)
But it's stuck here

**Comment by ws-choi** [comment]
I suspected overfitting due to the small size of the sample training dataset,
but it is the full dataset.

I also encountered NaN sometimes when training LaSAFT-based model, especially when training larger models.
It has quite sensitive hyperparameters to tune.

I checked the below script [works quite well](https://wandb.ai/wschoi/lasaft/reports/LaSAFT-larger-model-training-eval-example--Vmlldzo1NDczMTM), but it is a larger model (Table2 in our paper, n_fft=4096 version, need 4 gpus).

```
python main.py --problem_name conditioned_separation --mode train --musdb_root ../repos/musdb18_wav --n_blocks 9 --num_tdfs 6 --n_fft 4096 --hop_length 1024 --precision 16 --embedding_dim 64 --pin_memory True --save_top_k 3 --patience 10 --deterministic --model lasaft_net --gpus 4 --distributed_backend ddp --sync_batchnorm True --run_id lasaft_2021_als --batch_size 4 --seed 2021 --log wandb --lr 0.0001 --auto_lr_schedule True
```

I share what I've tried to avoid NaN issue.
- [ ] try larger batch size: try ```--batch_size 8``` instead of 6
- [ ] try smaller learning rate: If it is not possible to enlarge batch size, then try ```--lr 0.0005```
- [ ] rmsprop: ```--optimizer rmsprop``` was relatively robust for training LaSAFT
- [ ] deterministic: although ```--deterministic  Ture``` slows down training, but it helps NaN dubugging.
